lib/win32.js

```diff
@@ -0,0 +1,152 @@
+'use strict';
+
+const pify = require('pify');
+const fs = require('fs');
+const path = require('path');
+const readdir = pify(fs.readdir);
+const stat = pify(fs.stat);
+const fswin = require('fswin');
+
+/**
+ * Modify options object as needed for Windows
+ * @param opts Object
+ * @returns Object
+ */
+const handleOptions = opts => {
+  // Convert any CLI params to boolean
+  Object.keys(opts).forEach((key) => {
+    const val = opts[key];
+    if (typeof val === 'string') {
+      opts[key] = opts[key] === 'true';
+    }
+  });
+
+  if (opts.clear) {
+    opts.dependencyHidden = opts.devDependencyHidden = false;
+  }
+  return opts;
+};
+
+/**
+ * This hides all the dependencies first and writes the info file
+ * @param projectPath String
+ * @param opts Object
+ * @returns {Promise}
+ */
+const performPreAction = (projectPath, opts) => {
+  const modules = path.join(projectPath, 'node_modules');
+  const hidden = typeof opts.clear === 'boolean' ? !opts.clear : true;
+
+  fs.writeFile(
+    path.join(modules, 'modules_hidden.txt'),
+    'You\'ve hidden modules with `john` - https://github.com/davej/john',
+    {flags: 'w'},
+    function (err) {
+      if (err) {
+        console.warn('Failed to write info file', err);
```

Comment:
I prefer to only output to console in the CLI, console output isn't very useful when used programmatically. Perhaps we should just return an error? If we can't write a file to the `node_modules` folder then something is probably very wrong and it will fail when setting the hidden attribute anyway.

Also for consistency perhaps we can use `pify` on `fs.writeFile` and do a promise instead?